dspace-server-webapp/src/test/java/org/dspace/app/rest/csv/CSVMetadataImportReferenceTest.java

```diff
@@ -0,0 +1,328 @@
+        List<Relationship> rels = relationshipService.findByItem(context, leftItem);
+        int foundCount = 0;
+        for (Relationship rel : rels) {
+            if (rel.getLeftItem().getID().equals(leftItem.getID())
+                    && rel.getRightItem().getID().equals(rightItem.getID())) {
+                foundCount++;
+            }
+        }
+        assertEquals(expectedCount, foundCount);
+    }
+
+    /**
+     * Test existence of newly created items with proper relationships defined in mets via
+     * a metadata field and value reference
+     */
+    @Test
+    public void testSingleMdRef() throws Exception {
+        String[] csv = {"id,relationship.type,relation.isAuthorOfPublication,collection,dc.identifier.other",
+                "+,Person,," +  col1.getHandle() + ",0",
+                "+,Publication,dc.identifier.other:0," +  col1.getHandle() + ",1"};
+        Item[] items = runImport(csv);
+        assertRelationship(items[1], items[0], 1);
+    }
+
+    /**
+     * return an array of items given a representation of a CSV as a string array
+     *
+     * @param csvLines A representation of a CSV as a string array
+     *
+     * @return an array of items
+     */
+    private Item[] runImport(String[] csvLines) throws Exception {
+        performImportScript(csvLines, false);
+        Item[] items = new Item[csvLines.length - 1];
+        for (int i = 0; i < items.length; i++) {
+            items[i] =  itemService.findByIdOrLegacyId(context, getUUIDByIdentifierOther("" + i).toString());
+        }
+        return items;
+    }
+
+    /**
+     * Test existence of newly created item with proper relationships defined in the item's metadata via
+     * a rowName reference
+     */
+    @Test
+    public void testSingleRowNameRef() throws Exception {
+        String[] csv = {"id,dc.title,relationship.type,relation.isAuthorOfPublication,collection,rowName," +
+                "dc.identifier.other",
+                "+,Test Item 1,Person,," +  col1.getHandle() + ",idVal,0",
+                "+,Test Item 2,Publication,rowName:idVal," +  col1.getHandle() + ",anything,1"};
+        Item[] items = runImport(csv);
+        assertRelationship(items[1], items[0], 1);
+    }
+
+    /**
+     * Test existence of newly created items with proper relationships defined in the item's metadata via
+     * a metadata field and value reference with multiable references
+     */
+    @Test
+    public void testMultiMdRef() throws Exception {
+        String[] csv = {"id,relationship.type,relation.isAuthorOfPublication,collection,dc.identifier.other",
+        "+,Person,," +  col1.getHandle() + ",0",
+        "+,Person,," +  col1.getHandle() + ",1",
+        "+,Publication,dc.identifier.other:0||dc.identifier.other:1," +  col1.getHandle() + ",2"};
+        Item[] items = runImport(csv);
+        assertRelationship(items[2], items[0], 1);
```

Comment:
Can you also verify the order matches

Reply:
Yes, we will add an arg to assertRelationship and update tests to confirm correct order (place) is respected.